securedrop/tests/functional/test_source_session_timeout.py

```diff
@@ -1,17 +1,176 @@
-from . import source_navigation_steps
-from . import functional_test
+import socket
+import time
+from dataclasses import dataclass
+from multiprocessing import Process
+from pathlib import Path
 
+from selenium.webdriver.firefox.webdriver import WebDriver
 
-class TestSourceSessions(
-        functional_test.FunctionalTest,
-        source_navigation_steps.SourceNavigationStepsMixin):
+import pytest
+import requests
 
-    session_expiration = 5
+from models import Journalist
+from tests.functional.app_navigators import SourceAppNagivator, WebDriverTypeEnum, get_web_driver
+from tests.functional.db_session import get_database_session
+from tests.functional.factories import SecureDropConfigFactory
+from tests.functional.sd_config_v2 import SecureDropConfig
 
-    def test_source_session_timeout(self):
-        self._source_visits_source_homepage()
-        self._source_clicks_submit_documents_on_homepage()
-        self._source_continues_to_submit_page()
-        self._source_waits_for_session_to_timeout()
-        self.driver.refresh()
-        self._source_sees_session_timeout_message()
+
+def _get_unused_port() -> int:
+    s = socket.socket()
+    s.bind(("127.0.0.1", 0))
+    port = s.getsockname()[1]
+    s.close()
+    return port
+
+
+def _start_source_server(port: int, config: SecureDropConfig) -> None:
+    # This function will be called in a separate Process that runs the source app
+    # Modify the sdconfig module in the app's memory so that it mirrors the supplied config
+    # Do this BEFORE importing any other module of the application so the modified config is
+    # what eventually gets imported by the app's code
+    import sdconfig
+
+    sdconfig.config = config  # type: ignore
```

Comment:
The above is the "killer" feature of the new `sd_servers_with_short_session_timeout` fixture (which can become a more general `sd_servers_v2` fixture in a future PR): any SDConfig can be supplied, and the source and journalist apps to be spawned will use that (regardless of the content of securedrop/config.py or anything else).

Reply:
To make sure I understand correctly, all of this setup and fixture code is only in this file because this is the only test you've ported so far, in the end it would be shared across all functional tests and be in something like functional_test.py, right?

Reply:
Correct. We could also run the functional tests in parallel by spawning multiple sd_servers.

Reply:
I think it would make more sense to move these setup into conftest.py as part of this PR, as you're already doing in https://github.com/freedomofpress/securedrop/pull/6209/commits/8df04311dd254cced235449a66335a2c615f6d9a. Having it there from the beginning will be nicer since we're not moving the same bits of code around multiple times.